cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1831,9 +1832,12 @@ auto convert_table_to_parquet_data(table_input_metadata& table_meta,
   // Initialize data pointers in batch
   uint32_t const num_stats_bfr =
     (stats_granularity != statistics_freq::STATISTICS_NONE) ? num_pages + num_chunks : 0;
-  rmm::device_buffer uncomp_bfr(max_uncomp_bfr_size, stream);
-  rmm::device_buffer comp_bfr(max_comp_bfr_size, stream);
-  rmm::device_buffer col_idx_bfr(column_index_bfr_size, stream);
+
+  // Buffers need to be padded as the compute kernels require aligned data pointers.
+  rmm::device_buffer uncomp_bfr(cudf::detail::align_up(max_uncomp_bfr_size, 8 /*alignment*/),
+                                stream);
+  rmm::device_buffer comp_bfr(cudf::detail::align_up(max_comp_bfr_size, 8 /*alignment*/), stream);
+  rmm::device_buffer col_idx_bfr(column_index_bfr_size * 2, stream);
```

Comment:
Was this causing issues? `column_index_bfr_size` should already have alignment padding built in.

Reply:
Oh sorry, I tested with this variable, didn't find issue with it but forgot to change it back :smile: